### PR TITLE
ARSN-355 List lifecycle non-current versions supports V0

### PR DESCRIPTION
**Pull request by nicolas2bert**
This PR includes the following changes:

- The `evaluatedKeys` have been renamed to `scannedKeys`.
- The implementation, now, leverages `DelimiterVersions` to provide an accurate list of versions.
- The test has been updated to run on V0 and V1 bucket formats.

### CODE REVIEW

**Comment by nicolas2bert** [comment]
ping

File: lib/algos/list/delimiterNonCurrent.js
Comment (jonathan-gramain) [inline lib/algos/list/delimiterNonCurrent.js:21]: I think we should leave the jsdoc for `keyMarker` and `versionIdMarker` since those are still accepted by the parent class constructor. (there might be others as well).

File: lib/algos/list/delimiterNonCurrent.js
Comment (jonathan-gramain) [inline lib/algos/list/delimiterNonCurrent.js:39]: Don't we need to also set `start` in `genMDParamsV1`?

File: lib/algos/list/delimiterNonCurrent.js
Comment (jonathan-gramain) [inline lib/algos/list/delimiterNonCurrent.js:73]: Since the `DelimiterVersions` class exports the enum `DelimiterVersionsFilterStateId`, it should be possible to import it in this file and use its named values.

File: lib/algos/list/delimiterNonCurrent.js
Comment (jonathan-gramain) [inline lib/algos/list/delimiterNonCurrent.js]: Could this variable be clearer if named `isCurrentVersion`, like in the previous code? At this stage we don't know if the key being processed is a master or a version key, because this is left to the internal implementation of `DelimiterVersions` which contract is to return actual distinct versions to `addContents`, of which one of them is the current version, hence the suggestion.

For the same reason, it could be better to check if the version is the latest by checking if the `key` is different than the previous one (by saving the previous one in a class member) so to avoid using "internal" `this.masterKey` and `this.masterVersionId` fields from `DelimiterVersions`. I'm fine with keeping it as is if it helps, your call.

**Review by rachedbenmustapha — APPROVED** [review]

File: tests/unit/algos/list/delimiterNonCurrent.spec.js
Comment (jonathan-gramain) [inline tests/unit/algos/list/delimiterNonCurrent.spec.js]:
```suggestion
        it('should accept one delete marker and one version and return the noncurrent version', () => {
```

File: tests/unit/algos/list/delimiterNonCurrent.spec.js
Comment (jonathan-gramain) [inline tests/unit/algos/list/delimiterNonCurrent.spec.js]:
```suggestion
        it('should accept two versions and return the noncurrent version', () => {
```

File: tests/unit/algos/list/delimiterNonCurrent.spec.js
Comment (jonathan-gramain) [inline tests/unit/algos/list/delimiterNonCurrent.spec.js]:
```suggestion
        it('should accept three versions and return the noncurrent version which stale date before beforeDate', () => {
```

**Review by jonathan-gramain — APPROVED** [review]
approving with some minor comments

File: lib/algos/list/delimiterNonCurrent.js
Comment (nicolas2bert) [reply lib/algos/list/delimiterNonCurrent.js:39]: `genMDParamsV1` calls `genMDParamsV0`. 
There is a unit test that ensures `this.start` is always assigned for v0 and v1.

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
@bert-e approve

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
@bert-e reset

**Comment by nicolas2bert** [comment]
@bert-e reset

**Pull request merged**
